### DISCUSSION TITLE
How can I add a dispatching rule to the yaml file so as to get that immediately after I import the yaml file

### DISCUSSION THREAD

**Discussion by touensure** [microcks/microcks#969] [Q&A] [answered]
Hi, I want to add a dispatching rule that allows microcks to response according to a field in the request payload. I am not sure how to write in yaml code so that Mickrocks can work correctly without doing some manual adjustment after importing the  yaml file. Is there anybody who can help me with this question? Thanks a lot!

**Comment by lbroudoux** [comment, marked as answer]
Hi @touensure!

To dispatch incoming requests to mock response according a field in the request payload, you may want to use the `JSON_BODY` dispatcher. See [documentation](https://microcks.io/documentation/using/advanced/dispatching/#json-body-dispatcher).

You can embed this dispatcher specification directly into your OpenAPI file using our own Microcks extensions `x-microcks` so that the dispatching rule will be applied directly after your specification import. See [documentation](https://microcks.io/documentation/using/openapi/#using-openapi-extensions).

Hope it helps!

**Reply by touensure** [reply]
thanks a lot for your kind help!!
